cmd/hooks/prestop.go

```diff
@@ -0,0 +1,126 @@
+package hooks
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+)
+
+/*
+When a node is terminated, persistent workflows using EBS volumes can take 6+ minutes to start up again.
+This happens when a volume is not cleanly unmounted, which causes the Attach/Detach controller (in kube-controller-manager)
+to wait for 6 minutes before issuing a force detach and allowing the volume to be attached to another node.
+
+This PreStop lifecycle hook aims to ensure that before the node (and the CSI driver node pod running on it) is shut down,
+all VolumeAttachment objects associated with that node are removed, thereby indicating that all volumes have been successfully unmounted and detached.
+
+No unnecessary delay is added to the termination workflow, as the PreStop hook logic is only executed when the node is being drained
+(thus preventing delays in termination where the node pod is killed due to a rolling restart, or during driver upgrades, but the workload pods are expected to be running).
+If the PreStop hook hangs during its execution, the driver node pod will be forcefully terminated after 30 seconds (the default terminationGracePeriodSeconds for a pod).
+*/
+
+func PreStop(clientset kubernetes.Interface, timeout time.Duration) error {
+	klog.InfoS("PreStop: executing PreStop lifecycle hook")
```

Comment:
Suggestion: add `timeout` to this log message

Reply:
Done.